### DISCUSSION TITLE
About the Security groups

### DISCUSSION THREAD

**Discussion by leo79901** [apache/cloudstack#9943] [Q&A] [answered]
I have installed an ACS with 'Core - Basic' zone. When I want to change the VM's name, display or other information, ACS tells me that I cannot change the security groups.
Is this correct?
Meanwhile, can't I change the security groups while the VMs are running? If this is true, I have to create a new security group for every VM.

![image](https://github.com/user-attachments/assets/3d4ef15b-f553-4294-b207-949f682e56db)

**Comment by rajujith** [comment]
@leo79901 You should be allowed to change or display the name on a running instance. This error seems to be a bug with the basic zone only or any instance with a security group. Do you mind creating an issue ticket? 

You have to stop the instance to change the security group. 
You can use one security group for many instances.

**Reply by weizhouapache** [reply]
it looks like the restriction was added 8 years ago and it is still there

https://github.com/apache/cloudstack/commit/58772934344d2b0316ddb479a977b15b6e97c015

**Reply by leo79901** [reply]
Thanks a lot. In order to provide more flexible security protection, before it possible to adjust the security groups when the machine is powered on, I will create a separate security group for each host.

**Comment by NuxRo** [comment]
@leo79901 if you do it from cmk/cloudmonkey does it work?

**Reply by leo79901** [reply]
Yes, i have do it again using CloudMonkey. The name was changed successfully.
![image](https://github.com/user-attachments/assets/6c7161bb-5449-411f-b464-1d2ec4e6f889)

**Comment by weizhouapache** [comment, marked as answer]
in my opinion,

- the first issue is a bug. since security groups cannot be changed when vm is running, we should hide the field in the dialog.
- the second can be a improvement request: Update security group lists of a running vm

**Reply by leo79901** [reply]
Thanks a lot, i will do it.

**Reply by weizhouapache** [reply]
please keep eye on #10714 which aim to fix the first issue

**Reply by weizhouapache** [reply]
new feature request: #10729
